src/shared/mount-util.c

```diff
@@ -1119,11 +1139,31 @@ static int mount_in_namespace(
         if (r == 0) {
                 errno_pipe_fd[0] = safe_close(errno_pipe_fd[0]);
 
-                if (make_file_or_directory)
-                        (void) mkdir_p(dest, 0755);
+                if (img) {
+                        if (make_file_or_directory)
+                                (void) mkdir_p(dest, 0755);
```

Comment:
I think it would make a ton of sense to just do mkdir_parents() here (and thus move this line before the `if (img)` if block, and then add DISSECT_IMAGE_MKDIR to the mount flags below.